src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -393,18 +407,22 @@ internal override int CalculateLocalSyntaxOffset(int localPosition, SyntaxTree l
             throw ExceptionUtilities.Unreachable;
         }
 
-        internal override bool TryGetThisParameter(out ParameterSymbol thisParameter)
+        internal override bool TryGetThisParameter(out ParameterSymbol? thisParameter)
         {
             // Local function symbols have no "this" parameter
             thisParameter = null;
             return true;
         }
 
-        private static void ReportAttributesDisallowed(SyntaxList<AttributeListSyntax> attributes, DiagnosticBag diagnostics)
+        private void ReportAttributesDisallowed(SyntaxList<AttributeListSyntax> attributes, DiagnosticBag diagnostics)
         {
-            foreach (var attrList in attributes)
+            var diagnosticInfo = MessageID.IDS_FeatureLocalFunctionAttributes.GetFeatureAvailabilityDiagnosticInfoOpt(ScopeBinder.Compilation);
```

Comment:
>ScopeBinder.Compilation [](start = 118, length = 23)

Consider using `DeclaringCompilation` instead for consistency with other uses.

Reply:
So, this needs a look, because DeclaringCompilation is null for some reason in EE scenarios. Perhaps that is a bug?

Reply:
filed #39165 to follow up on this.